### DISCUSSION TITLE
How to set the "target_session_attrs" attribute for postgresql connection ?

### DISCUSSION THREAD

**Discussion by articque** [doctrine/DoctrineBundle#1660] [Q&A] [answered]
Hello,

I'm working on a symfony project with postgreqsql and multiple hosts for database. There is primary server and a replicated server.
In my configuration, i set two IP in doctrine.dbal.connections.default.host, separated by coma. Example : 192.10.11.12,192.10.11.13
192.10.11.12 is my primary server, 192.10.11.13 is my replicated server.
And it works. Even if my primary server crashes and the replicated become the new primary, it always works.

But if i invert the two IP, it doesn't work because all the requests reach the replicated server (the first IP) which is only in read-only mode.
I need to say to postgresql : "send requests only on my primary server" (it doesn't matter if it's the 1st or the 2nd IP)
It's why I need to specify the "target_session_attrs", in my case, with the value "primary". 

Here the pgsql documentation for this attribute : https://www.postgresql.org/docs/current/libpq-connect.html

1/ Is there an option to set this attribute ?
2/ Otherwise, is there a way to set extra attributes to the final connection strings (example add "?target_session_attrs=primary") ?

Thanks in advance.

**Comment by ostrolucky** [comment, marked as answer]
This is not supported because it's not supported by DBAL, see here https://github.com/doctrine/dbal/blob/3.6.x/src/Driver/PgSQL/Driver.php You should add this feature request there.

You are always free to create your own instance of the AbstractPostgreSQLDriver where you can put to `pg_connect` whatever you want though.
